### PR TITLE
AB#904 Create autocomplete component

### PR DESCRIPTION
**Pull request by mkidawa**

### CODE REVIEW

File: src/common/components/Autocomplete/Autocomplete.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    <MuiAutocomplete
+      ref={ref}
+      className={className}
+      sx={{ color: 'grey.800' }}
+      multiple
+      disableCloseOnSelect
+      inputValue={inputValue}
+      open={isOpen}
+      onOpen={() => {
+        setIsOpen(true);
```
Comment (JKrymarys) [inline src/common/components/Autocomplete/Autocomplete.tsx:74]: Meybe let's take out `onOpen` and `onClose` handlers from return?

File: src/common/components/Autocomplete/Autocomplete.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+          <span>{label}</span>
+          <TextField
+            {...params}
+            InputProps={{
+              ...params.InputProps,
+              ref: params.InputProps.ref,
+              placeholder: placeholder,
+              autoFocus: true,
+              sx: {
+                padding: '5px 40px !important',
```
Comment (JKrymarys) [inline src/common/components/Autocomplete/Autocomplete.tsx:103]: Just to double check. Is `!important` here on purpose? Are we sure that it's unavoidable ?

File: src/common/components/Autocomplete/Autocomplete.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                borderRadius: 0,
+                border: 'none',
+                '& fieldset': {
+                  borderTop: 'none',
+                  borderLeft: 'none',
+                  borderRight: 'none',
+                  borderBottom: '1.5px solid grey.400 !important',
```
Comment (JKrymarys) [inline src/common/components/Autocomplete/Autocomplete.tsx:110]: Ditto. Can't we void `!important`?

File: src/common/components/Autocomplete/Autocomplete.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        setInputValue(newInputValue);
+      }}
+      value={value}
+      onChange={(event, newValue: AutocompleteOption[]) => {
+        if (handleValueChange) {
+          handleValueChange(newValue);
+        }
+        setOptions(options);
+        setValue(newValue);
+      }}
```
Comment (JKrymarys) [inline src/common/components/Autocomplete/Autocomplete.tsx:134]: What about extracting those functions to the body of the component, and keep return as short/concise as possible?

File: src/common/components/Autocomplete/Autocomplete.tsx [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+                </Box>
+              );
+            })}
+          </Box>
+        </li>
+      )}
+      options={options}
+      noOptionsText="Brak wyników"
+      getOptionLabel={(option) => option.primaryLabel}
+      filterOptions={(x) => x}
```
Comment (JKrymarys) [inline src/common/components/Autocomplete/Autocomplete.tsx:167]: Is this necessary ? Isn't that just the default behavior?

File: src/common/components/Autocomplete/Autocomplete.tsx (hunk 2 above)
Comment (mkidawa) [reply src/common/components/Autocomplete/Autocomplete.tsx:103]: Unfortunetaly yes :/ Tried both ways

File: src/common/components/Autocomplete/Autocomplete.tsx (hunk 5 above)
Comment (mkidawa) [reply src/common/components/Autocomplete/Autocomplete.tsx:167]: That is necessary :P It overrides default filtering

File: src/common/components/Autocomplete/Autocomplete.tsx (hunk 3 above)
Comment (mkidawa) [reply src/common/components/Autocomplete/Autocomplete.tsx:110]: Ye, that one could have been avoided :D

File: src/common/components/Autocomplete/Autocomplete.tsx (hunk 5 above)
Comment (mkidawa) [reply src/common/components/Autocomplete/Autocomplete.tsx:165]: obraz

**Review by szymonbla — APPROVED** [review]

File: src/common/components/Autocomplete/Autocomplete.tsx (hunk 5 above)
Comment (JKrymarys) [reply src/common/components/Autocomplete/Autocomplete.tsx:165]: oh god xd

**Review by JKrymarys — APPROVED** [review]
looks good 👍🏻

**Pull request merged**